Secure postgres connections

ndegwamartin opened the pull request:
Fixes [Issue 2046 ](https://github.com/onaio/playbooks/issues/2046)

ndegwamartin committed:
Add support for configuration of secure postgres db connections

Co-authored-by: Morris Mukiri <mmukiri@ona.io>

ndegwamartin committed:
Download CA Certificate Bundles
Add support for downloading and installing CA Certificate bundles

ndegwamartin committed:
Set Postgres ssl mode verification to Verify-CA

ndegwamartin committed:
Fix formatting

ndegwamartin committed:
Update variables
  Add configuration variables
   - opensrp_postgres_enable_ssl
   - opensrp_postgres_sslmode

  rename downloaded ca bundle registered variable

jasonrogena commented on roles/opensrp/tasks/main.yml:
```diff
@@ -63,10 +63,26 @@
+  become_user: root
+  get_url:
+    url: "{{ ca_bundle_url }}"
+    dest: "/usr/local/share/ca-certificates/ca-bundle.crt"
```
@ndegwamartin this isn't a deal breaker, but do you think it'd be a good idea to rename the file `ca-bundle.crt` to something that is more reflective that this is related to OpenSRP and database access to prevent a collision?

ndegwamartin replied:
Yeah that makes sense. so like **_db-ca-bundle.crt_**?

ndegwamartin replied:
Or **_opensrp-db-ca-bundle.crt_** ?

ndegwamartin replied:
I've updated this to be **postgres-ca-bundle.crt** as per our discussion earlier

jasonrogena commented on roles/opensrp/tasks/main.yml:
```diff
@@ -63,10 +63,26 @@
+  become: yes
+  become_user: root
+  get_url:
+    url: "{{ ca_bundle_url }}"
```
@ndegwamartin This should follow the naming convention used in the roles. So maybe `opensrp_ca_bundle_url`.

ndegwamartin replied:
This is updated too

ndegwamartin committed:
Update postgres ca bundle download scripts
 - Refactor to constraint postgres ca bundle set up commands to postgres task only

ndegwamartin committed:
Consistent naming of variables
 - Update naming convention to one used for the OpenSRP roles

ndegwamartin committed:
logic error
 - Fixes logical error for update postgres ca certificate condition task
